substrate/bin/utils/chain-spec-builder/src/lib.rs

```diff
@@ -385,15 +397,45 @@ impl CreateCmd {
 	}
 }
 
-/// Processes `CreateCmd` and returns string represenataion of JSON version of `ChainSpec`.
+/// Parses chain properties passed as a comma-separated KEY=VALUE pairs.
+fn parse_properties(raw: &String, props: &mut sc_chain_spec::Properties) -> Result<(), String> {
+	for pair in raw.split(',') {
+		let mut iter = pair.splitn(2, '=');
+		let key = iter.next().ok_or("Invalid chain property key")?.trim().to_owned();
```

Comment:
```suggestion
		let key = iter.next().ok_or_else(|| format!("Invalid chain property key: {pair}"))?.trim().to_owned();
```

Reply:
Addressed in [d9b1292](https://github.com/paritytech/polkadot-sdk/pull/7368/commits/d9b12928af1d77ae1549ec743e741595a92d8914).